lib/matplotlib/axes/_axes.py

```diff
@@ -3284,6 +3284,18 @@ def errorbar(self, x, y, yerr=None, xerr=None,
         if len(x) != len(y):
             raise ValueError("'x' and 'y' must have the same size")
 
+        def check_if_negative(array):
+            if array is None:
+                return False
+            try:
+                if np.any(array < 0):
+                    return True
+            except TypeError:   # Don't fail on 'datetime.*' types
+                pass
+
+        if check_if_negative(xerr) or check_if_negative(yerr):
+            raise ValueError("'xerr' and 'yerr' must have positive numbers")
```

Comment:
Wording (a value of 0 is ok):

> ... must not contain negative numbers.

or 

> ... must have non-negative numbers.

Reply:
good catch! Thanks! Also fixed it in the description